src/main/java/seedu/realodex/logic/parser/DeleteCommandParser.java

```diff
@@ -12,21 +12,45 @@
  */
 public class DeleteCommandParser implements Parser<DeleteCommand> {
 
+    public static final String INDEX_AND_NAME_PROVIDED = "Please provide either an index or a name, not both.";
+    public static final String NO_FIELDS_PROVIDED = "Please provide either an index or a name.";
+
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME);
+        String trimmed = args;
+
+        // Parse name if present
+        Name name;
         if (namePrefixPresent(argMultimap)) {
-            Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-            return new DeleteCommand(name);
-        } else {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
+            name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+            trimmed = args.replace("n/" + name.fullName, "");
         }
 
+        // Parse index if present
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(trimmed);
+        } catch (ParseException pe) {
+            index = null;
```

Comment:
not too sure about this, it looks like exceptions are used as control flow. anyone else agree?

Reply:
I think its ok following from `EditCommandParser`

Reply:
this is very similar to udhaya's pr for #140 and my pr for #154. I was thinking about it but seems that to check whether `trimmed` can be parsed, we need to parse it first